### PR TITLE
Add script to generate documentation

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: scripts/generate-docs.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# This script will generate documentation for the current version
+
+VERSION=`python3 -c "import json; f = open('./package.json', 'r'); v = json.loads(f.read())['version']; f.close(); print(v)"`
```
Comment (jaller94) [inline scripts/generate-docs.sh:4]: > Relying on python3 bin in a node package

```suggestion
VERSION=`/usr/bin/env node -e "console.log(require('./package.json').version)"`
```

Also, there should be some `if` and `exit` for if this fails.

File: scripts/generate-docs.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+git checkout gh-pages
+mv .typedoc/ $VERSION/
+sed "6i\    <li><a href=\"$VERSION/index.html\">$VERSION</a></li>" index.html > index.html.new
+rm index.html
+mv index.html.new index.html
+
+git add "$VERSION/"
+git add index.html
+git commit -m "Add documentation for $VERSION"
+git push
```
Comment (jaller94) [inline scripts/generate-docs.sh:20]: When using automated commits and pushs some checks should have been made that the preceding commands did the expected thing (e.g. not fail or actually create the files/folders).

**Review by jaller94 — CHANGES_REQUESTED** [review]
I'm super excited to get the documentation going, but this lacks some conditions in the bash script.
The cost of this script to go haywire isn't much but it should at least stop and exit with an error code if VERSION doesn't start with a number, yarn gendoc's return code (`$?`) isn't 0 and the branch `gh-pages` is not get checked out after the command.

File: scripts/generate-docs.sh (hunk 1 above)
Comment (Half-Shot) [reply scripts/generate-docs.sh:4]: Hah, fair points

File: scripts/generate-docs.sh (hunk 2 above)
Comment (Half-Shot) [reply scripts/generate-docs.sh:20]: Should be covered by line 6? It will fail the script if any of these return non zero.

**Review by jaller94 — APPROVED** [review]
Yay for auto-generating the docs.

**Pull request merged**